script/tool/lib/src/format_command.dart

```diff
@@ -212,4 +241,65 @@ class FormatCommand extends PluginCommand {
 
     return javaFormatterPath;
   }
+
+  /// Returns true if [command] can be run successfully.
+  Future<bool> _hasDependency(String command) async {
```

Comment:
I don't know if we have minimum version restrictions for java/clang-formant, but this looks like a cool point to add that validation when required! Good stuff!

Reply:
I'm not sure for Java; for clang-format we should actually have an exact version per https://github.com/flutter/flutter/issues/39767

I can see about fixing that as a follow-up while I'm thinking of it.